server/models/components/utils.py

```diff
@@ -1,59 +1,75 @@
 from scipy.optimize import fsolve
-
-
-def compute_ratio(dms, max_days):
-    def f(q):
-        return (1 / 2) * (1 - q ** max_days) * (1 + q) / (1 - q) - dms
-
-    solutions = fsolve(f, 5)
-    ratio = solutions[0]
-    return ratio
-
-
-def compute_residuals(dms, max_days):
-    ratio = compute_ratio(dms, max_days)
-    residuals = [ratio ** i for i in range(max_days + 1)]
+import matplotlib.pyplot as plt
+
+
+def factorial(k) :
+    
+    fact = 1
+    
+    for i in range(1, k + 1) :
+        fact *= i
+    
+    return fact
+
+def binomial_coeff(k, n) :
+    
+    return factorial(n) / (factorial(k) * factorial(n - k))
+
+def compute_param(dms, max_days, prob_type):
+    
+    if prob_type == 'exponential':
+        def f(x):
+            return (1 - x ** (max_days)) / (1 - x) - dms
+        solutions = fsolve(f, 5)
+        param = solutions[0]
+    elif prob_type == 'binomial':
+        param = (dms - 1) / max_days
+        
+    return param
+
+def compute_khi(dms, max_days = 21, prob_type = 'binomial') :
+    
+    khi_tab = []
+    param = compute_param(dms, max_days, prob_type)
+    
+    if prob_type == 'exponential':
```

Comment:
wouldn't it be better to introduce 2 different functions, one for  exponential and another one for binomial.
This is not so much code-sharing in the current state